android/java/org/chromium/chrome/browser/crypto_wallet/AssetRatioServiceFactory.java

```diff
@@ -19,22 +19,22 @@
 
 @JNINamespace("chrome::android")
 public class AssetRatioServiceFactory {
-    private static final Object lock = new Object();
-    private static AssetRatioServiceFactory instance;
+    private static final Object sLock = new Object();
+    private static AssetRatioServiceFactory sInstance;
 
     public static AssetRatioServiceFactory getInstance() {
-        synchronized (lock) {
-            if (instance == null) {
-                instance = new AssetRatioServiceFactory();
+        synchronized (sLock) {
```

Comment:
I see this pattern is used a lot, so this is a suggestion for a future improvement: we can get rid of the `sLock` object completely, this would also save up memory resources as we don't instantiate a new object.
To synchronize the access we simply add the `synchronized` keyword in the method: `public static synchronized AssetRatioServiceFactory getInstance()`